File: android/app/src/main/java/com/colageclub/colage/features/auth/screens/ServerTypeScreen.kt
package com.colageclub.colage.features.auth.screens

import androidx.compose.animation.*
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import com.colageclub.colage.core.design.*
import com.colageclub.colage.data.models.ServerType
import com.colageclub.colage.features.auth.AuthViewModel

@Composable
fun ServerTypeScreen(
    authViewModel: AuthViewModel,
    onContinue: () -> Unit
) {
    var selected by remember { mutableStateOf<ServerType?>(null) }

    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(ColageColors.Background)
    ) {
        Column(
            modifier = Modifier.fillMaxSize(),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            Spacer(Modifier.weight(1f))

            Text(
                text = "Are you a…",
                style = ColageFonts.LargeTitle.copy(color = ColageColors.TextPrimary),
                textAlign = TextAlign.Center
            )

            Spacer(Modifier.height(8.dp))

            Text(
                text = "This determines which server you join",
                style = ColageFonts.Subheadline.copy(color = ColageColors.TextSecondary),
                textAlign = TextAlign.Center
            )

            Spacer(Modifier.height(32.dp))

            ServerTypeCard(
                icon = Icons.Default.MenuBook,
                title = "Current Student",
                description = "Join your school's server with other students on campus",
                isSelected = selected == ServerType.STUDENT,
                onTap = { selected = ServerType.STUDENT },
                modifier = Modifier.padding(horizontal = 24.dp)
            )

            Spacer(Modifier.height(16.dp))

            ServerTypeCard(
                icon = Icons.Default.School,
                title = "Alumni",
                description = "Join the Alumni Network — graduates from all schools, one community",
                isSelected = selected == ServerType.ALUMNI,
                onTap = { selected = ServerType.ALUMNI },
                modifier = Modifier.padding(horizontal = 24.dp)
            )

            Spacer(Modifier.weight(1f))

            Button(
                onClick = {
                    selected?.let { type ->
                        authViewModel.updateServerType(type)
                        onContinue()
                    }
                },
                enabled = selected != null,
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 24.dp)
                    .padding(bottom = 50.dp)
                    .height(56.dp),
                shape = RoundedCornerShape(16.dp),
                colors = ButtonDefaults.buttonColors(
                    containerColor = ColageColors.Primary,
                    disabledContainerColor = ColageColors.Primary.copy(alpha = 0.3f)
                )
            ) {
                Text(
                    "Continue",
                    style = ColageFonts.BodyBold.copy(color = Color.White)
                )
            }
        }
    }
}

@Composable
private fun ServerTypeCard(
    icon: ImageVector,
    title: String,
    description: String,
    isSelected: Boolean,
    onTap: () -> Unit,
    modifier: Modifier = Modifier
) {
    Row(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(16.dp))
            .background(ColageColors.Surface)
            .then(
                if (isSelected) Modifier.border(2.dp, ColageColors.Primary, RoundedCornerShape(16.dp))
                else Modifier
            )
            .clickable(onClick = onTap)
            .padding(16.dp),
        verticalAlignment = Alignment.CenterVertically
    ) {
        Box(
            modifier = Modifier
                .size(56.dp)
                .clip(CircleShape)
                .background(if (isSelected) ColageColors.Primary.copy(alpha = 0.2f) else ColageColors.Surface),
            contentAlignment = Alignment.Center
        ) {
            Icon(
                icon,
                contentDescription = null,
                tint = if (isSelected) ColageColors.Primary else ColageColors.TextSecondary,
                modifier = Modifier.size(28.dp)
            )
        }

        Spacer(Modifier.width(16.dp))

        Column(modifier = Modifier.weight(1f)) {
            Text(title, style = ColageFonts.BodyBold.copy(color = ColageColors.TextPrimary))
            Spacer(Modifier.height(4.dp))
            Text(description, style = ColageFonts.Caption.copy(color = ColageColors.TextSecondary))
        }

        Spacer(Modifier.width(12.dp))

        Icon(
            if (isSelected) Icons.Default.CheckCircle else Icons.Default.RadioButtonUnchecked,
            contentDescription = null,
            tint = if (isSelected) ColageColors.Primary else ColageColors.TextTertiary,
            modifier = Modifier.size(24.dp)
        )
    }
}

File: Colage/Features/Auth/Screens/ServerTypeScreen.swift
import SwiftUI

/// Onboarding screen where the user picks Student or Alumni
struct ServerTypeScreen: View {
    let onContinue: () -> Void
    @Environment(.themeColor) private var themeColor
    @EnvironmentObject var onboardingData: OnboardingData

    @State private var selected: ServerType? = nil
    @State private var animateIn = false

    var body: some View {
        ZStack {
            ColageColors.background.ignoresSafeArea()

            VStack(spacing: 0) {
                Spacer()

                VStack(spacing: 24) {
                    Text("Are you a…")
                        .font(ColageFonts.largeTitle)
                        .foregroundStyle(ColageColors.textPrimary)
                        .opacity(animateIn ? 1 : 0)

                    Text("This determines which server you join")
                        .font(ColageFonts.subheadline)
                        .foregroundStyle(ColageColors.textSecondary)
                        .opacity(animateIn ? 1 : 0)

                    VStack(spacing: 16) {
                        ServerTypeCard(
                            type: .student,
                            icon: "book.fill",
                            title: "Current Student",
                            description: "Join your school's server with other students on campus",
                            isSelected: selected == .student,
                            onTap: { selected = .student }
                        )

                        ServerTypeCard(
                            type: .alumni,
                            icon: "graduationcap.fill",
                            title: "Alumni",
                            description: "Join the Alumni Network — graduates from all schools, one community",
                            isSelected: selected == .alumni,
                            onTap: { selected = .alumni }
                        )
                    }
                    .padding(.horizontal, 24)
                    .opacity(animateIn ? 1 : 0)
                }

                Spacer()

                ColagePrimaryButton(title: "Continue", action: {
                    guard let type = selected else { return }
                    onboardingData.serverType = type
                    onContinue()
                })
                .disabled(selected == nil)
                .opacity(selected != nil ? 1 : 0.5)
                .padding(.horizontal, 24)
                .padding(.bottom, 50)
                .animation(.easeInOut(duration: 0.2), value: selected)
            }
        }
        .navigationBarBackButtonHidden(false)
        .onAppear {
            withAnimation(.spring(response: 0.6, dampingFraction: 0.8).delay(0.1)) {
                animateIn = true
            }
        }
    }
}

// MARK: - Server Type Card

private struct ServerTypeCard: View {
    let type: ServerType
    let icon: String
    let title: String
    let description: String
    let isSelected: Bool
    let onTap: () -> Void

    var body: some View {
        Button(action: onTap) {
            HStack(spacing: 16) {
                ZStack {
                    Circle()
                        .fill(isSelected ? themeColor.opacity(0.2) : ColageColors.surface)
                        .frame(width: 56, height: 56)

                    Image(systemName: icon)
                        .font(.system(size: 24))
                        .foregroundStyle(isSelected ? themeColor : ColageColors.textSecondary)
                }

                VStack(alignment: .leading, spacing: 4) {
                    Text(title)
                        .font(ColageFonts.bodyBold)
                        .foregroundStyle(ColageColors.textPrimary)

                    Text(description)
                        .font(ColageFonts.caption)
                        .foregroundStyle(ColageColors.textSecondary)
                        .multilineTextAlignment(.leading)
                }

                Spacer()

                Image(systemName: isSelected ? "checkmark.circle.fill" : "circle")
                    .font(.system(size: 24))
                    .foregroundStyle(isSelected ? themeColor : ColageColors.textTertiary)
            }
            .padding(16)
            .background(
                RoundedRectangle(cornerRadius: 16)
                    .fill(ColageColors.surface)
                    .overlay(
                        RoundedRectangle(cornerRadius: 16)
                            .stroke(isSelected ? themeColor : Color.clear, lineWidth: 2)
                    )
            )
        }
        .buttonStyle(.plain)
    }
}

#Preview {
    NavigationStack {
        ServerTypeScreen(onContinue: {})
            .environmentObject(OnboardingData())
    }
}
